sql/opt/exec/execbuilder: TestExecBuild failed

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2054938&tab=buildLog) on [master@84c0a3e960421ec754583daa708c8c3542b4f1e3](https://github.com/cockroachdb/cockroach/commits/84c0a3e960421ec754583daa708c8c3542b4f1e3):

```
=== RUN   TestExecBuild/local/srfs/correlated_SRFs
rewrote:
CREATE TABLE xy (x INT8 PRIMARY KEY, y INT8, FAMILY (x, y));


rewrote:
CREATE TABLE xz (x INT8 PRIMARY KEY, z INT8, FAMILY (z, x));


rewrote:
CREATE TABLE groups (id SERIAL8, data JSONB, PRIMARY KEY (id), FAMILY (id, data));


rewrote:
CREATE TABLE articles (id INT8 PRIMARY KEY, body STRING, description STRING, title STRING, slug STRING, tag_list STRING[], user_id STRING, created_at TIMESTAMP, updated_at TIMESTAMP, FAMILY (id, created_at, slug), FAMILY (user_id, title), FAMILY (body, tag_list), FAMILY (description, updated_at));


rewrote:
CREATE TABLE a (x INT8 PRIMARY KEY, y INT8, FAMILY (y, x));
CREATE TABLE b (x INT8 PRIMARY KEY, z INT8, FAMILY (z), FAMILY (x));


--- done: testdata/srfs with config local: 22 tests, 0 failures
--- total progress: 240 statements/queries

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2108456&tab=buildLog) on [master@e9a4f83e3eee59510f97db2c6e0df9b57cf6b944](https://github.com/cockroachdb/cockroach/commits/e9a4f83e3eee59510f97db2c6e0df9b57cf6b944):

```
=== RUN   TestExecBuild
    TestExecBuild: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild990370903
    TestExecBuild: test_log_scope.go:58: use -show-logs to present logs inline
--- FAIL: TestExecBuild (44.84s)
=== RUN   TestExecBuild/5node/stats
=== PAUSE TestExecBuild/5node/stats
=== CONT  TestExecBuild/5node/stats
--- progress: testdata/distsql_interleaved_join: 40 statements/queries
    TestExecBuild/5node/stats: logic.go:2062: 
         
        testdata/stats:22: EXPLAIN (VERBOSE) SELECT * FROM uv WHERE u = 1 AND v = 1
        expected:
            ·          distribution  full         ·       ·
            ·          vectorized    true         ·       ·
            filter     ·             ·            (u, v)  ·
             │         filter        u = 1        ·       ·
             └── scan  ·             ·            (u, v)  ·
            ·          table         uv@uv_v_idx  ·       ·
            ·          spans         /1-/2        ·       ·
            
        but found (query options: "") :
            ·          distribution  full         ·       ·
            ·          vectorized    true         ·       ·
            filter     ·             ·            (u, v)  ·
             │         filter        v = 1        ·       ·
             └── scan  ·             ·            (u, v)  ·
            ·          table         uv@uv_u_idx  ·       ·
            ·          spans         /1-/2        ·       ·
            
--- done: testdata/stats with config 5node: 5 tests, 1 failures
    TestExecBuild/5node/stats: logic.go:2643: 
        testdata/stats:33: error while processing
    TestExecBuild/5node/stats: logic.go:2643: testdata/stats:33: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/stats (1.18s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/opt/exec/execbuilder).TestExecBuild failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2109242&tab=buildLog) on [master@dafb69aa4af50d965698738323680c9175d59511](https://github.com/cockroachdb/cockroach/commits/dafb69aa4af50d965698738323680c9175d59511):

```
=== RUN   TestExecBuild
    TestExecBuild: logic.go:2762: randomize batchSize to 3
    TestExecBuild: test_log_scope.go:77: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestExecBuild063628629
    TestExecBuild: test_log_scope.go:58: use -show-logs to present logs inline
--- FAIL: TestExecBuild (63.01s)
=== RUN   TestExecBuild/5node/stats
=== PAUSE TestExecBuild/5node/stats
=== CONT  TestExecBuild/5node/stats
        --- FAIL: TestExecBuild/5node/stats (2.46s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestExecBuild.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/opt/exec/execbuilder/testdata/stats
+++ b/pkg/sql/opt/exec/execbuilder/testdata/stats
@@ -15,11 +15,26 @@ statement ok
 CREATE STATISTICS u ON u FROM uv;
 CREATE STATISTICS v ON v FROM uv
 
+query TTIIIB colnames
+SELECT
+  statistics_name,
+  column_names,
+  row_count,
+  distinct_count,
+  null_count,
+  histogram_id IS NOT NULL AS has_histogram
+FROM
+  [SHOW STATISTICS FOR TABLE uv]
+----
+statistics_name  column_names  row_count  distinct_count  null_count  has_histogram
+u                {u}           8          2               0           true
+v                {v}           8          7               0           true
+
 statement ok
 set enable_zigzag_join = false
 
 # Verify we scan index v which has the more selective constraint.
-query TTTTT
+query TTTTT retry
 EXPLAIN (VERBOSE) SELECT * FROM uv WHERE u = 1 AND v = 1
 ----
 ·          distribution  full         ·       ·
@@ -301,7 +316,7 @@ limit
 statement ok
 CREATE STATISTICS tj FROM tj
 
-query T
+query T retry
 EXPLAIN (OPT, VERBOSE) SELECT DISTINCT j FROM tj WHERE j IS NULL
 ----
 limit

